accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/aem/AemCqActionsSupportImpl.java

```diff
@@ -8,60 +8,253 @@
 import java.util.Map;
 import java.util.Set;
 
 import javax.jcr.Node;
 import javax.jcr.RepositoryException;
 import javax.jcr.Session;
+import javax.jcr.Value;
+import javax.jcr.nodetype.NodeDefinition;
+import javax.jcr.nodetype.NodeType;
+import javax.jcr.security.AccessControlManager;
 import javax.jcr.security.Privilege;
 
+import org.apache.jackrabbit.JcrConstants;
+import org.apache.jackrabbit.api.security.JackrabbitAccessControlList;
+import org.apache.jackrabbit.api.security.JackrabbitAccessControlManager;
 import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import com.day.cq.security.util.CqActions;
+import biz.netcentric.cq.tools.actool.configmodel.AceBean;
+import biz.netcentric.cq.tools.actool.helper.AccessControlUtils;
 
 @Component(service = AemCqActionsSupport.class)
 public class AemCqActionsSupportImpl implements AemCqActionsSupport {
+    private static final Logger LOG = LoggerFactory.getLogger(AemCqActionsSupportImpl.class);
 
-    public static class CqActionsWrapper implements AemCqActions {
-        
-        private final CqActions cqActions;
+    public enum CqActions {
+        read, modify, create, delete, acl_read, acl_edit, replicate;
+    }
 
-        public CqActionsWrapper(com.day.cq.security.util.CqActions cqActions) {
-            this.cqActions = cqActions;
-        }
+    public static final String REPLICATE_PRIVILEGE = "{http://www.day.com/crx/1.0}replicate";
 
-        public Collection<String> getAllowedActions(String nodePath, Set<Principal> principals) throws RepositoryException {
-            return cqActions.getAllowedActions(nodePath, principals);
-        }
 
-        public void installActions(String nodePath, Principal principal, Map<String, Boolean> actionMap, Collection<String> inheritedAllows) throws RepositoryException {
-            cqActions.installActions(nodePath, principal, actionMap, inheritedAllows);
-        }
-
-        @SuppressWarnings("deprecation")
-        public Set<Privilege> getPrivileges(String action) {
-            return cqActions.getPrivileges(action);
-        }
-    }
-    
     public AemCqActions getCqActions(Session session) {
-    
-        CqActions cqActions;
+
         try {
-            cqActions = new CqActions(session);
-            final CqActionsWrapper cqActionWrapper = new CqActionsWrapper(cqActions);
-            return cqActionWrapper;
+            return new AcToolCqActions(session);
         } catch (Exception e) {
-            throw new IllegalStateException("Could not instantiate CqActions: "+e, e);
+            throw new IllegalStateException("Could not instantiate CqActions: " + e, e);
         }
     }
-    
-    
+
     public boolean definesContent(Node node) throws RepositoryException {
-        return CqActions.definesContent(node);
+        return AcToolCqActions.definesContent(node);
     }
     
+    public static class AcToolCqActions implements AemCqActions {
```

Comment:
was this taken over from some AEM class? If so please add some comment from where/which version